### PR TITLE
Adds the HECU gas mask & associated sounds

### PR DESCRIPTION
**Pull request by T-Wiz**
Adds the HECU gas mask, which attempt to convert anything said by the wearer into HECU vox.
Adds soldier vox sounds from everyone's favorite resonance cascade simulator.
Adds the HECU gas mask to the Trader Alcatraz bundle.
image

This is my first PR and first time really messing with ss13 code, so criticism is appreciated.

:cl:
 * rscadd: The Trader's Alcatraz bundle now has a new HECU gas mask.
 * soundadd: MY. ASS. IS. HEAVY.

### CODE REVIEW

**Comment by Buchse** [comment]
MY.
ASS.
IS.
HEAVY.

**Comment by ShiftyRail** [comment]
HECU is just a filter to the side and a flat-ish mask covering the face. Your sprite looks like you have two filters, which look kinda weird.
The oculars of the mask alone look off. They're perfect square when the original mask is mores slinted.

Good job on adding the mask, though.

XBx1UounR0LU

File: code/modules/clothing/masks/gasmask.dm [hunk 1]
```diff
@@ -320,4 +320,92 @@
...
+		return
+	return togglevoice(usr)
+
+/obj/item/clothing/mask/gas/hecu/attack_self()
+	src.togglevoice()
+	..()
+	return
+
+/obj/item/clothing/mask/gas/hecu/proc/togglevoice(var/mob/user = usr)
+	if(usr.isUnconscious())
```
Comment (ShiftyRail) [inline code/modules/clothing/masks/gasmask.dm]: Replace `usr` by `user`. `usr` is a byond internal variable which we prefer not to use.

File: code/modules/clothing/masks/gasmask.dm [hunk 2]
```diff
@@ -320,4 +320,92 @@
...
+		return
+	else
+		togglestate = !togglestate
+
+/datum/action/item_action/toggle_voice //blank on purpose
```
Comment (ShiftyRail) [inline code/modules/clothing/masks/gasmask.dm]: Could add a name and description.

**Review by ShiftyRail — COMMENTED** [review]
Personally I would prefer them for the Deathsquad than a traitor.

**Comment by tuypo1** [comment]
Now I can upload a bunch of futa to the library and recreate that one cut scene from the game

File: code/modules/clothing/masks/gasmask.dm [hunk 3]
```diff
@@ -320,4 +320,92 @@
...
+	processing_objects.Add(src)
+
+/obj/item/clothing/mask/gas/hecu/Destroy()
+	processing_objects.Remove(src)
+	..()
+
+obj/item/clothing/mask/gas/hecu/process()
+	if(mask_charge >= max_charge)
+		return
+	else
```
Comment (boy2mantwicethefam) [inline code/modules/clothing/masks/gasmask.dm]: Else is not necessary

File: code/modules/clothing/masks/gasmask.dm [hunk 4]
```diff
@@ -320,4 +320,92 @@
...
+		mask_charge++
+		return
```
Comment (boy2mantwicethefam) [inline code/modules/clothing/masks/gasmask.dm]: Return is not necessary

**Comment by boy2mantwicethefam** [comment]
Man I was supposed to do the HECU gas masks like a few months later!

File: code/modules/clothing/masks/gasmask.dm [hunk 5]
```diff
@@ -320,4 +320,92 @@
...
+
+/obj/item/clothing/mask/gas/hecu/attack_self()
+	src.togglevoice()
+	..()
+	return
+
+/obj/item/clothing/mask/gas/hecu/proc/togglevoice(var/mob/user = usr)
+	if(usr.isUnconscious())
+		return
+	else
```
Comment (boy2mantwicethefam) [inline code/modules/clothing/masks/gasmask.dm]: You don't need else

File: code/modules/clothing/masks/gasmask.dm [hunk 6]
```diff
@@ -320,4 +320,92 @@
...
+			for(var/j=1,j<=hecuwords.len,j++)
+				if (lowertext(hecuwords[j]) == lowertext(word_list[i]))
+					words_to_say += hecuwords[j]
+		if(words_to_say.len > 0)
+			for(var/i=1,i<=words_to_say.len,i++)
+				if(mask_charge >= word_cost)
+					mask_charge -= word_cost
+					playsound(src, "sound/vox_hecu/[words_to_say[i]]!.wav", 30)
+					sleep word_delay
+		else return
```
Comment (boy2mantwicethefam) [inline code/modules/clothing/masks/gasmask.dm]: you don't need an else return
Also try to replace else return with ..()

**Comment by T-Wiz** [comment]
I couldn't figure out how to get the HUD button to work, I think it's because gas masks already have one? Anyway, SHOUTING now activates the mask; with a side benefit of letting it be used in the middle of a sentence.

**Pull request merged**